### PR TITLE
#82 - Added django debug toolbar

### LINKED ISSUE #82: Setup a dev specific config that includes django-debug-toolbar.
It would be highly useful to have the [django-debug-toolbar](https://github.com/jazzband/django-debug-toolbar) available when running the django app for development purposes.

We would not want it loaded as part of the typical configuration that would get run on deployed instances, so we should instead setup a debug configuration such that we can set `DJANGO_SETTINGS_MODULE=functionary.settings.debug` and have that configuration effectively load the standard config and then override the necessary settings.

Alternatively, we could have a bunch of checks against the `DEBUG` setting inline in the main `django_.py` settings file, but that could get ugly fast if we have additional settings that we want to override over time.

### PR DESCRIPTION
**Pull request by xeonneon**
Closes #82 

This PR adds the Django Debug Toolbar as part of the new debug settings module. Settings modules now inherit from a `base.py` module, which instantiates all the default values. A separate module file can then import all the module-level variables from `base.py` and overwrite them as necessary.

## Testing
- In `docker-compose.yml`, edit the `DJANGO_SETTINGS_MODULE` to point to whichever settings file you want to use
- If using the `debug` settings module, visit Functionary's UI and verify the Django Debug Toolbar is available on the right-side of the screen

### CODE REVIEW

**Review by scott-taubman — CHANGES_REQUESTED** [review]

File: functionary/requirements.in [hunk 1]
```diff
@@ -1,5 +1,6 @@
 celery
 django
+django-debug-toolbar
```
[the remark this answers is not in the document]
Comment (xeonneon) [reply functionary/requirements.in]: Resolved.

When doing the `pip-compile` to remake the requirements.txt, some packages pulled in their minor patches. Is this expected whenever we update the requirements.in?

File: functionary/functionary/settings/debug.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+DEBUG = True
+
+INSTALLED_APPS += ["debug_toolbar"]
+
+# Note: Order of MIDDLEWARE content matters
+MIDDLEWARE += ["debug_toolbar.middleware.DebugToolbarMiddleware"]
+
+
+# Docker specific method for setting INTERNAL_IPS
+if DEBUG:
```
[the remark this answers is not in the document]
Comment (xeonneon) [reply functionary/functionary/settings/debug.py]: Resolved.

File: functionary/requirements.in (hunk 1 above)
[the remark this answers is not in the document]
Comment (scott-taubman) [reply functionary/requirements.in]: Yeah, that's not a problem.

File: functionary/functionary/settings/__init__.py [hunk 3]
```diff
@@ -1,8 +1,10 @@
+from .base import *  # noqa
 from .builder_ import *  # noqa
 from .celery_ import *  # noqa
 from .core_ import *  # noqa
-from .django_ import *  # noqa
+from .debug import *  # noqa
```
[the remark this answers is not in the document]
Comment (xeonneon) [reply functionary/functionary/settings/__init__.py]: Resolved.

I was running into issues where if I was not exporting all the modules, some settings were not being seen by the Django apps. Everything is working fine without them now, so I suspect I misconfigured something and forgot to re-test.

File: functionary/functionary/settings/debug.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    import socket  # only if you haven't already imported this
```
[the remark this answers is not in the document]
Comment (xeonneon) [reply functionary/functionary/settings/debug.py]: Resolved.

File: functionary/functionary/settings/prod.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+from .base import BASE_DIR
```
[the remark this answers is not in the document]
Comment (xeonneon) [reply functionary/functionary/settings/prod.py]: Resolved.

Was a remnant from when I was trying to work around Flake8's error messages.

**Review by scott-taubman — APPROVED** [review]

**Pull request merged**